training/templates/training/register.html

```diff
@@ -1,174 +1,187 @@
-    {% bootstrap_field form.website %}
-    {% bootstrap_field form.location %}
-</div>
-
-<h2>Estimating Resource Usage</h2>
-
-<p>
-This is an <b>EXTREMELY IMPORTANT</b> section for us and for you, the more detailed you can be,
-the better we can be in granting you the correct resources. If we only grant resources that are too
-small, then you will be stuck waiting in the normal queue because the jobs cannot be scheduled on
-the VMs we have allocated for you.
-</p>
-
-<p>
-Tool resource usage can be very surprising. We schedule tools based on their WORST CASE MEMORY
-USAGE. We do not have many high-memory machines. These two facts combined mean that sometimes a tool
-can use an expected amount of memory when running on our infrastructure.
-</p>
-
-<p>
-If you can provide the precise workflows you will run, that is optimal. We can extract the list
-of tools and compare that against our job configuration, then we can allocate a VM with enough
-resources for your participants to run that tool (on a small dataset) in a reasonable amount of
-time.
-</p>
-
-<div class="editable">
-    {% bootstrap_field form.use_gtn %}
-    {% bootstrap_field form.gtn_links%}
-    {% bootstrap_field form.non_gtn_links%}
-    {% bootstrap_field form.attendance %}
-</div>
-
-<h2>Training Identifier</h2>
 
-<p>
-We need a short keyword to name your training. It should be a single word, and can contain
-numbers and letters. Once the infrastructure is setup, you'll receive a URL like 
-{{ settings.TIAAS_DOMAIN }}/join-training/your-keyword
-</p>
+{% block head %}
+<link href="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/css/select2.min.css" rel="stylesheet" />
+<link rel="stylesheet" href="{% static 'training/css/register.css' %}">
+{% endblock %}
 
-<p>
-On the day of your training (or before), you'll tell your attendees to go to this URL and it will
-automatically place them in a group that redirects them to the private resources dedicated for your
-training
-</p>
 
-<div class="editable">
-    {% bootstrap_field form.training_identifier %}
-</div>
+{% block content %}
 
-{% if settings.TIAAS_SHOW_ADVERTISING %}
-<h2>Advertising</h2>
+<!--
+django-bootstrap template tag reference:
+https://django-bootstrap3.readthedocs.io/en/latest/templatetags.html
+-->
+
+<div class="container">
+	<h2>Apply for a new training allocation</h2>
+
+	<div class="my-5">
+		{% include 'snippets/register/1_intro.html' %}
+	</div>
 
-<p>
-We offer this service for free so we would love to be able to show the positive impacts providing
-it has had in the worldwide Galaxy community.
-</p>
+	<form action="{% url 'register' %}" method="post" class="form">
+		{% csrf_token %}
 
-<div class="editable">
-    {% bootstrap_field form.advertise_eu %}
-    {% bootstrap_field form.blogpost %}
+		<div>
+			{% bootstrap_field form.name placeholder="" form_group_class="form-group short-field" %}
+			{% bootstrap_field form.email placeholder="" form_group_class="form-group short-field" %}
```

Comment:
I still don't feel comfortable with such bare form fields. I really want some more explanation describing why we ask for this data. Explaining that the email will be redacted after a period of time (based on retention settings) is useful to know. Also reminding users that the Name field doesn't need to be their legal name, and it's just used for addressing you, for me at least (:transgender_flag:), I find to be an important aspect

Reply:
I'll pass this back to the team but I have two suggestions for now:

- Finding a way to template (customise) the form fields that isn't painful
- Add some info tooltips so this additional information is available, but not crowding the form

In our experience the vast majority of users don't read any of the instructions or details that we present them, so we try to include only the most relevant information to make the form as clear as possible. It's all a balance between being clear and comprehensive.

Reply:
Just chatting with Gareth - we'd be happy to put in a few tooltips with the data usage info. We could also change the name label to "preferred name"? Whatever communicates this in the best and clearest way.

On the other hand, if you'd like have more control over the form content as a whole, we can look into more granular templating instead. We could have a YAML file which allows the deployer to set `label` `help` `tooltip` variables for each field. That makes the form totally configurable.

Reply:
Yeah "preferred name" would communicate the same thing! That'd be fine there.

I don't think I want a yaml solution, or really customisability at all, I want sensible defaults. In practice there's a whole 5 deployments, no one is actually going to reconfigure what text is shown on the label, it's too niche of a setting. Tooltips are a good compromise here!

Reply:
Awesome, we can definitely work with that :) Are there other areas where you think the default needs more?

Reply:
I did have another chat with Gareth about the email redaction thing - our angle was that we often communicate with tiaas clients through our email and ticketing system. So in order to honestly say that we redact their contact information, we'd have to purge the information from those systems as well (which can't be done in an automated way like tiaas).

So I've updated the email field to show this help text:

![image](https://user-images.githubusercontent.com/42562517/192389002-65d6940a-743a-45b3-818d-a91f754f8644.png)

Which I hope works for everybody (otherwise please suggest changes).

Then the configurable `retain_contact` field can be used to display the GDPR statement, which is currently:

> If you consent we will retain your information for a longer period of time (XX months). We will use this to contact you regarding letters of support for our continued funding, and similar matters.

Reply:
No, that's perfect, I like it!